packages/postcss-reduce-initial/src/script/__tests__/io.js

```diff
@@ -0,0 +1,86 @@
+import nock from 'nock';
+import { handleError, toJSONString, write, generate } from '../lib/io.mjs';
+import testData from './sampleProperties.json';
+
+describe('toJSONString', () => {
+  const rawData = {
+    foo: 'bar',
+    'baz-qux': 'quux',
+    quuz: 'corge',
+    garply: 'waldo',
+    'fred-plugh': 'xyzzy-thud',
+  };
+
+  test('should have correct data type (string)', () => {
+    expect(typeof toJSONString(rawData)).toBe('string');
+  });
+
+  test('should produce parsable JSON', () => {
+    expect(JSON.parse(toJSONString(rawData))).toEqual(rawData);
+  });
+
+  test('should have a trailing newline', () => {
+    expect(toJSONString(rawData)).toMatch(/\n$/);
+  });
+});
+
+describe('File operations', () => {
```

Comment:
I also see this suite as a bit superfluous. It tests the `write()` interface but we don't really care about it since in the production code `fileFunc` is always the same.  Similarly it tests that the file paths are configurable, but we don't  care since in fact the plugin requires a specific name (`fromInitial` and `toInitial`).

Reply:
It verifies that `write` passes the data through `toJSONString` when calling `writeFiles` - as for the configurable file paths, they are a result of other considerations that I have tried to explain [here](https://github.com/cssnano/cssnano/pull/1160/files#r667018251). The comment [here](https://github.com/cssnano/cssnano/pull/1160/files#r666879350) is also relevant in regards to how the test fits in with others.

I included `should handle file operation errors` for the sake of coverage and the fact that `fs.writeFile` will call this callback regardless of whether there is an error or not... so it should only throw when there is an error passed in.